File: atcoder.jp/Python (CPython 3.11.4)/abc318/abc318_b.py
N = int(input())

sheet = [[0] * 100 for _ in range(100)]

for _ in range(N):
    X1, X2, Y1, Y2 = map(int, input().split())
    for i in range(Y1, Y2):
        for j in range(X1, X2):
            sheet[i][j] += 1

ans = 0
for i in range(100):
    for j in range(100):
        if sheet[i][j] > 0:
            ans += 1

print(ans)

File: atcoder.jp/Rust (rustc 1.70.0)/abc318/abc318_b.rs
use proconio::{input, fastout};

#[fastout]
fn main() {
    input! {
        n: usize,
    }

    let mut sheet = vec![vec![0; 101]; 101];
    for _ in 0..n {
        input! {
            a: usize,
            b: usize,
            c: usize,
            d: usize,
        }
        for x in a..b {
            for y in c..d {
                sheet[x][y] = 1;
            }
        }
    }

    let ans = sheet.iter().flatten().sum::<i32>();
    println!("{}", ans);
}
